pkg/core/tekton/tekton.go

```diff
@@ -454,3 +498,88 @@ func getInputCodesetPath(inputs []*workflow.WorkflowStepInput) string {
 	}
 	return ""
 }
+
+func (w *WorkflowBackend) toWorkflowRun(wf workflow.Workflow, p v1beta1.PipelineRun) *workflow.WorkflowRun {
+	startTime := p.Status.StartTime.Format(time.RFC3339)
+	wfr := workflow.WorkflowRun{
+		Name:        &p.ObjectMeta.Name,
+		WorkflowRef: &wf.Name,
+		StartTime:   &startTime,
+	}
+
+	if p.Status.CompletionTime != nil {
+		completionTime := p.Status.CompletionTime.Format(time.RFC3339)
+		wfr.CompletionTime = &completionTime
+	}
+
+	for _, input := range wf.Inputs {
+		var value string
+		if *input.Type == inputTypeCodeset {
+			value = fmt.Sprintf("%s:%s", *getPipelineResourceParamsValue("url", p.Spec.Resources[0]),
+				*getPipelineResourceParamsValue("revision", p.Spec.Resources[0]))
+		} else {
+			value = *getPipelineRunParamValue(*input.Name, p.Spec.Params)
+		}
+		wfr.Inputs = append(wfr.Inputs, &workflow.WorkflowRunInput{
+			Input: input,
+			Value: &value,
+		})
+	}
+
+	for _, output := range wf.Outputs {
+		wfr.Outputs = append(wfr.Outputs, &workflow.WorkflowRunOutput{
+			Output: output,
+			Value:  getPipelineRunResultValue(*output.Name, p.Status.PipelineResults),
+		})
+	}
+	status := "Unknown"
+	if len(p.Status.Conditions) > 0 {
+		status = pipelineReasonToWorkflowStatus(p.Status.Conditions[0].Reason)
+	}
+	wfr.Status = &status
+
+	url := fmt.Sprintf("%s/#/namespaces/%s/pipelineruns/%s", w.dashboardURL, w.namespace, *wfr.Name)
+	wfr.URL = &url
+	return &wfr
+}
+
+func pipelineReasonToWorkflowStatus(reason string) string {
```

Comment:
This is more for me than a review comment: this trim function is needed because some status values have `PipelineRun` as a prefix (see https://github.com/tektoncd/pipeline/blob/main/docs/pipelineruns.md#monitoring-execution-status).

BTW, it would be nice if you added this as a function comment ^ :)

Reply:
comment added!

Reply:
thanks